### PR TITLE
Rhys/markojs

### PR DESCRIPTION
**Pull request by wheresrhys**
Early days WIP

A problem I'm wrestling wit, which you may have thoughts on, is that marko has no inherited scope other than the global one. i.e, in handlebars, partials inherit all the variables in the parent scope. In marko they must be explicitly passed in, and accessed as properties on `input` (feels like there should be a way to skip `input.`)

This is kinda fine, if a little annoying at times, but when one introduces the notion of a presenter class it feels clunky. i.e `input.presenter.propertyUsingGetter`. It's be nice to extend/convert `input` to become the presenter. Feel free to branch off this WIP to have a play around

### CODE REVIEW

**Comment by kavanagh** [comment]
OOoooooooh

**Comment by wheresrhys** [comment]
A marko teaser: 
image

**Comment by wheresrhys** [comment]
image

File: helpers/n-image.js [hunk 1]
```diff
@@ -0,0 +1,170 @@
...
+ * @param {string[]|string} [classes = []] - Additional classes to add to the img element
+ * @param {string[]|string} [wrapperClasses = []] - Additional classes to add to the wrapper element
+ * @param {string} [width] - Width of the image
+ * @param {string} [height] - Height of the image
+ * @param {string} [alt = ''] - Alt text for the image
+ * @param {boolean} [lazyLoad = false] - Lazy load the image
+ * @param {string} [sourceParam = 'next'] - The `source` parameter used in the image service request
+ * @param {string|number} [placeholder] - Set to add a placeholder. Value should be the ratio of the image as a number
+ * (width divided by height, e.g.`16/9`), or string (`square` or `landscape`)
+ */
```
Comment (i-like-robots) [inline helpers/n-image.js]: this is terrifying

**Comment by i-like-robots** [comment]
In general I think this is looking like an improvement =]

**Comment by wheresrhys** [comment]
image

**Pull request closed without merging**